api/v1alpha1/virtualmachineimage_types.go

```diff
@@ -41,6 +41,16 @@ type VirtualMachineImageOSInfo struct {
 	Type string `json:"type,omitempty"`
 }
 
+// VirtualMachineImageInfo describes basic information related to the image, such as id and name. This information
+// can be used to identify an image in the content source.
+type VirtualMachineImageInfo struct {
```

Comment:
Other should chime in but my preference would be to move these two fields into the Spec. Arguably the whole Spec is already "info", and I think somebody looking at the currently proposed Spec would wonder why other "info" - OSInfo, ProductInfo, and the other Spec fields - aren't under this "info".

Reply:
I think `ImageID` should definitely be in spec, since it's immutable and will be used to uniquely correlate the vmimage with cl item. (this is what the design doc also specified)

However, I think `ImageName` is fine to be in status (since it can change over time). But you're right, the OSInfo/ProductInfo should have probably been done in status as well (since they can also change over time, in case vi admin uploads a new version of the CL Item).

Reply:
I agree that we erred and much of current Spec should have been placed in the Status. That said, I think my preference for now would be to to group similar things together, even if that isn't preferred location. I totally see the other side of let's add new fields to the preferred location going forward, so I don't either have a strong option. Although IDK for a later v1a2 the conversion hooks become a lot more complicated - or just not possible - if moving a field between the Spec and Status subresource.